client/trino-jdbc/src/test/java/io/trino/jdbc/TestTrinoDriverAuth.java

```diff
@@ -256,6 +256,78 @@ public void testSuccessFullSslVerification()
         }
     }
 
+    @Test
+    public void testSuccessFullSslVerificationAlternateHostname()
+            throws Exception
+    {
+        String accessToken = newJwtBuilder()
+                .setSubject("test")
+                .setHeaderParam(KEY_ID, "33")
+                .signWith(privateKey33)
+                .compact();
+
+        String url = format("jdbc:trino://127.0.0.1:%s", server.getHttpsAddress().getPort());
+        Properties properties = new Properties();
+        properties.setProperty("SSL", "true");
+        properties.setProperty("SSLVerification", "FULL");
+        properties.setProperty("SSLTrustStorePath", new File(getResource("localhost.truststore").toURI()).getPath());
+        properties.setProperty("SSLTrustStorePassword", "changeit");
+        properties.setProperty("accessToken", accessToken);
+        properties.setProperty("hostnameInCertificate", "localhost");
```

Comment:
Can you please also provide a test where connection fails without this option?

Reply:
Done!